client/components/login-window/index.jsx

```diff
@@ -0,0 +1,164 @@
+					fill="white"
+				/>
+			</svg>
+		);
+	};
+
+	const serviceData = {
+		wordpress: {
+			cookieName: 'wpc_wpc',
+			name: 'WordPress.com',
+			popup: ',height=980,width=500',
+			icon: WordPressIcon,
+			class: 'wordpress-login',
+		},
+	};
+
+	useEffect( () => {
+		setUserInfo( getUserInfoCookie() );
+	}, [] );
+
+	const logout = () => {
+		const logoutURL = getLogoutUrl( userInfo );
+		const serviceName = userInfo?.service;
+		const cookieName = serviceData[ serviceName ].cookieName;
+
+		const cookie = `${ cookieName }=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/; SameSite=None; Secure=True;${ addWordPressDomain }`;
+		document.cookie = cookie;
+
+		if ( serviceName === 'wordpress' ) {
+			// Atomic logging out
+			if ( window.location.host === 'jetpack.wordpress.com' ) {
+				const newURL =
+					logoutURL + '&redirect_to=' + window.location.hash.match( /#parent=(.*)/ )[ 1 ];
+				window.parent.location.href = newURL;
+			} else {
+				window.location.href = logoutURL + '&redirect_to=' + window.location.href;
+			}
+		}
+	};
+
+	const { res } = useQuery( {
+		queryKey: [ 'verbum/auth' ],
+		queryFn: () =>
+			wpcom.req.get( {
+				path: `/verbum/auth`,
+				apiNamespace: 'wpcom/v2',
+			} ),
+		select: ( data ) => {
+			//console.log( data );
+			return data;
+		},
+	} );
+	//console.log( 'res: ', res );
+	setUserInfo( res );
+
+	const login = async ( service ) => {
+		const { connectURL } = 'https://wordpress.com/log-in';
+
+		const loginWindow = window.open(
+			`${ connectURL }&service=${ service }`,
+			'VerbumCommentsLogin',
+			`status=0,toolbar=0,location=1,menubar=0,directories=0,resizable=1,scrollbars=0${ serviceData[ service ].popup }`
+		);
+
+		// Listen for login data
+		window.addEventListener( 'message', function waitForLogin( event ) {
+			if ( event.origin !== document.location.origin ) {
+				return;
+			}
+
+			if ( event.data.service === service ) {
+				setUserInfo( event.data );
+
+				if ( ! document.cookie.includes( 'wpc_' ) ) {
```

Comment:
hmm what would happen if the user has an expired login cookie? is that a thing?

Reply:
No cookies needed - taken out